src/sentry/mediators/sentry_app_installations/authorizer.py

```diff
@@ -49,14 +49,24 @@ def _validate_sentry_app(self):
 
     def _validate_grant(self):
         if (
-            self.grant.application.owner != self.user or
-            self.grant.application.client_id != self.client_id
+            not self.grant_belongs_to_install() or
+            not self.application_owned_by_user() or
+            not self.client_id_matches()
```

Comment:
I would likely order these so the cheapest to execute checks happen first and the ones that emit multiple database queries happen later. It could save us some queries in bad actor/spammer scenarios.

Reply:
I don't know if these matter anymore, with the addition of the list of `select_related` statements below.

Each of these functions uses one of those prefetched objects, so this entire condition should result in a single query now.